discovery/rendezvous.go

```diff
@@ -0,0 +1,159 @@
+package discovery
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/rand"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/p2p/discover"
+	"github.com/ethereum/go-ethereum/p2p/discv5"
+	"github.com/ethereum/go-ethereum/p2p/enr"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/status-im/rendezvous"
+)
+
+const (
+	registrationPeriod = 10 * time.Second
+	bucketSize         = 10
+)
+
+func NewRendezvous(servers []ma.Multiaddr, identity *ecdsa.PrivateKey, node *discover.Node) (*Rendezvous, error) {
+	r := new(Rendezvous)
+	r.servers = servers
+	r.registrationPeriod = registrationPeriod
+	r.bucketSize = bucketSize
+
+	r.record = enr.Record{}
+	r.record.Set(enr.IP(node.IP))
+	r.record.Set(enr.TCP(node.TCP))
+	r.record.Set(enr.UDP(node.UDP))
```

Comment:
I don't see the public key in this ENR record. Is it added by `enr.SignV4()`?

Reply:
yes

Reply:
👍 We should add a comment with an explanation.